### PR TITLE
Custom flags and context and stuff

### PR DESCRIPTION
**Pull request by wizjany**
Lets us do some more advanced stuff and lets other plugins register flags and session handlers.

### CODE REVIEW

File: src/main/java/com/sk89q/worldguard/bukkit/WorldGuardPlugin.java [hunk 1]
```diff
@@ -37,10 +47,33 @@
...
+import com.sk89q.worldguard.bukkit.listener.WorldGuardBlockListener;
+import com.sk89q.worldguard.bukkit.listener.WorldGuardCommandBookListener;
+import com.sk89q.worldguard.bukkit.listener.WorldGuardEntityListener;
+import com.sk89q.worldguard.bukkit.listener.WorldGuardHangingListener;
+import com.sk89q.worldguard.bukkit.listener.WorldGuardPlayerListener;
+import com.sk89q.worldguard.bukkit.listener.WorldGuardServerListener;
+import com.sk89q.worldguard.bukkit.listener.WorldGuardVehicleListener;
+import com.sk89q.worldguard.bukkit.listener.WorldGuardWeatherListener;
+import com.sk89q.worldguard.bukkit.listener.WorldGuardWorldListener;
+import com.sk89q.worldguard.bukkit.listener.WorldRulesListener;
```
Comment (TomyLobo) [inline src/main/java/com/sk89q/worldguard/bukkit/WorldGuardPlugin.java]: I like it!
Better to explicitly list all the imports so there are no surprises.

**Comment by ZephireNZ** [comment]
Is this likely to be pulled? Seems like a worthy addition for plugins to use

**Comment by TomyLobo** [comment]
All I know is it says "[WIP]"

**Comment by ZephireNZ** [comment]
@wizjany Can you explain what parts are WIP? I might be able to help finish it off, I've used it for custom flags but had to custom merge with older builds to get it working.

**Pull request merged**